components/dashboard/src/projects/new-project/NewProjectCreateFromURL.tsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC, useCallback, useMemo } from "react";
+import isURL from "validator/lib/isURL";
+import { CreateProjectArgs } from "../../data/projects/create-project-mutation";
+import { Subheading } from "../../components/typography/headings";
+import { Button } from "../../components/Button";
+import { useToast } from "../../components/toasts/Toasts";
+
+type Props = {
+    repoSearchFilter: string;
+    isCreating: boolean;
+    onCreateProject: (args: CreateProjectArgs) => void;
+};
+export const NewProjectCreateFromURL: FC<Props> = ({ repoSearchFilter, isCreating, onCreateProject }) => {
+    const { toast } = useToast();
+    const showCreateFromURL = useMemo(() => {
+        // TODO: Only accounts for https urls, need to account for ssh clone urls too?
+        const looksLikeURL = isURL(repoSearchFilter, {
+            require_protocol: true,
+            protocols: ["https"],
+        });
+
+        const hasTrailingGit = /\.git$/.test(repoSearchFilter);
+
+        return looksLikeURL && hasTrailingGit;
+    }, [repoSearchFilter]);
+
+    const normalizedURL = useMemo(() => {
+        let url = repoSearchFilter.toLowerCase().trim();
+
+        return url;
+    }, [repoSearchFilter]);
+
+    const handleCreate = useCallback(() => {
+        let name = "";
+        let slug = "";
+
+        try {
+            // try and parse the url for owner/repo path parts
```

Comment:
We've a util for that in server. I could make sense to pull it out to gitpod-protocol for reuse. Have a look at the special cases, some of them actually introduced for BBS:
https://github.com/gitpod-io/gitpod/blob/de09b5f828e23b8dfb6774c750c94b2db846c00d/components/server/src/repohost/repo-url.ts

Reply:
<img width="831" alt="Screenshot 2023-08-17 at 15 31 02" src="https://github.com/gitpod-io/gitpod/assets/914497/4e17fec0-e476-4caf-9d74-8c433a6725c4">

And this request params show, something is off.

Reply:
Thanks for catching, it's updated to use the repo name correctly now, and the slug becomes a mashup of the "owner" path parts and the repo name.

<img width="501" alt="image" src="https://github.com/gitpod-io/gitpod/assets/367275/b1582f86-fb00-4e3a-8540-1bfacb35dc8a">